src/datasources/balances-api/zerion-balances-api.service.ts

```diff
@@ -147,7 +147,7 @@ export class ZerionBalancesApi implements IBalancesApi {
       );
       return this._mapBalances(chainName, zerionBalances.data);
     } catch (error) {
-      if (error instanceof LimitReachedError) {
+      if (error instanceof LimitReachedError || error instanceof ZodError) {
```

Comment:
Moving this check into the `HttpErrorFactory` seems excessive as it is only used in a few select places.

Reply:
Thanks for your explanation. Isn't it used in all the places where we validate API responses? If that's the case I would move it to the factory.

Reply:
Generally, we validate API responses on the domain layer. It is only used in a select few places (touched in this PR) to traverse from a `Raw<T>` type to `T`. I don't think it's therefore worthwhile locating it in the factory. What do you think?